src/sage/schemes/elliptic_curves/hom.py

```diff
@@ -924,6 +924,102 @@ def is_zero(self):
         """
         return not self.degree()
 
+    def is_cyclic(self):
+        r"""
+        Determine whether the isogeny is cyclic (separable with cyclic kernel).
+
+        This is a general method, it requires factorization of the degree
+        and computation on the kernel polynomial (if not squarefree).
+        For special isogeny implementations (Frobenius, scalar multiplication,
+        composite) more efficient methods are used.
+
+        ALGORITHM:
+
+        Separable isogenies with squarefree degrees are cyclic.
+        For each $\ell \ne p$ such that its square divides the degree of the isogeny,
+        test if $E[\ell]$ is mapped to 0 by reducing the kernel polynomial
+        modulo the division polynomial. If for all $\ell$ it does not happen,
+        then the result is cyclic.
+
+        EXAMPLES:
+
+        Here we include also special cases which do not use the general algorithm. ::
```

Comment:
```suggestion
        EXAMPLES::
```
I think just showing the examples is fine